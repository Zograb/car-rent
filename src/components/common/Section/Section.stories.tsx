import type { Meta, StoryObj } from '@storybook/react';

import { Section } from './Section';

const meta: Meta<typeof Section> = {
  title: 'Section',
  component: Section,
  tags: ['autodocs'],
};

type Story = StoryObj<typeof Section>;

export default meta;

export const Basic: Story = {
  args: {
    children: <span>Section content</span>,
  }
};
