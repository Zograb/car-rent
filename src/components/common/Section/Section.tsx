import type { PropsWithChildren, FC, ComponentPropsWithoutRef } from 'react';
import 'twin.macro';

const Section: FC<PropsWithChildren<ComponentPropsWithoutRef<'div'>>> = ({
  children,
  ...otherProps
}) => (
  <div
    tw="py-8 px-7 mt-5 shadow-section rounded-lg"
    {...otherProps}
    data-testid="section"
  >
    {children}
  </div>
);

export default Section;
