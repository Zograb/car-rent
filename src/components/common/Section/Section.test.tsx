import { render, screen } from '@testing-library/react';

import Section from './Section';

describe('Section', () => {
  it('should match snapshot', () => {
    const { container } = render(<Section />);
    expect(container).toMatchSnapshot();
  });

  it('should render section', () => {
    render(
      <Section>
        <span>Section content</span>
      </Section>
    );
    const section = screen.getByText('Section content');
    expect(section).toBeInTheDocument();
  });
});

