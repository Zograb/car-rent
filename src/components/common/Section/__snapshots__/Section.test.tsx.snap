// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should match snapshot 1`] = `
.c0 {
  margin-top: 1.25rem;
  border-radius: 0.5rem;
  padding-left: 1.75rem;
  padding-right: 1.75rem;
  padding-top: 2rem;
  padding-bottom: 2rem;
  --tw-shadow: 0px 8px 24px rgba(69, 69, 80, 0.1);
  --tw-shadow-colored: 0px 8px 24px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow);
}

<div>
  <div
    class="c0"
    data-testid="section"
  />
</div>
`;
